### PR TITLE
Jennyf/netcore5

### PR DESCRIPTION
**Pull request by jennyf19**
- [x] there are 6 tests failing w/net 5, getting a null ref. will be investigating that, but thought best to work from this branch

 @jmprieur

### CODE REVIEW

File: build/template-install-dotnet-core.yaml [hunk 1]
```diff
@@ -5,6 +5,6 @@ steps:
 - task: UseDotNet@2
   displayName: 'Use .Net Core SDK'
   inputs:
-    version: 3.1.101, 5.0.100-preview.5.20279.10
+    version: 5.0.100-preview.5.20279.10
```
Comment (bgavrilMS) [inline build/template-install-dotnet-core.yaml:8]: yeah, I was suprised that you can have a list there :)

Comment (jennyf19) [reply build/template-install-dotnet-core.yaml]: that's what the documentation says to do. idk. doesn't seem to work.

Comment (jennyf19) [reply build/template-install-dotnet-core.yaml:8]: @bgavrilMS had to make separate tasks for each TFW

**Comment by jennyf19** [comment]
@jmprieur this is ready now. will need to update the release build when the time comes to .net 5

File: tests/Microsoft.Identity.Web.Test/WebApiExtensionsTests.cs [hunk 2]
```diff
@@ -99,9 +99,7 @@ private void AddProtectedWebApi_TestCommon(IServiceCollection services, ServiceP
             Assert.Equal(ServiceLifetime.Singleton, services.First(s => s.ServiceType == typeof(IJwtBearerMiddlewareDiagnostics)).Lifetime);
 
             // JWT options added correctly
-            var configuredJwtOptions = provider.GetService<IConfigureOptions<JwtBearerOptions>>() as ConfigureNamedOptions<JwtBearerOptions>;
-
-            Assert.Equal(_jwtBearerScheme, configuredJwtOptions.Name);
+            var configuredJwtOptions = provider.GetService<IConfigureOptions<JwtBearerOptions>>() as IConfigureNamedOptions<JwtBearerOptions>;
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Web.Test/WebApiExtensionsTests.cs:102]: - This is an issue w/.NET 5, it's a `ConfigureNamedOptions<JwtBearerOptions, Microsoft.AspNetCore.Authentication.JwtBearer>`, which is not a `ConfigureNamedOptions<JwtBearerOptions>. and, Microsoft.AspNetCore.Authentication.JwtBearer` is not a public type.
 
`x.GetType().FullName
"Microsoft.Extensions.Options.ConfigureNamedOptions`2[[Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerOptions, Microsoft.AspNetCore.Authentication.JwtBearer, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.IServiceProvider, System.ComponentModel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]]"`

We probably don't need to check the `name`, this isn't surfaced by the interface.

**Pull request merged**